aten/src/ATen/native/cpu/Reduce.h

```diff
@@ -118,9 +118,10 @@ static inline void vectorized_outer_reduction(char** data, int64_t inner_stride,
 template<typename traits, typename res_t>
 static void set_result(const int index, const res_t result, const TensorIteratorBase &iter, const int num_outputs) {
   // static_assert(std::is_same<res_t, typename traits::arg2_t>::value, "data types must match");
```

Comment:
I was going to suggest, shouldn't `out_t` be the same as `res_t`, but I guess not. When does it fail?